### PR TITLE
[DomCrawler] Document XPath expression evaluation

### PR DESCRIPTION
**Pull request by jakzal**
Documents https://github.com/symfony/symfony/pull/19430

### CODE REVIEW

File: components/dom_crawler.rst [hunk 1]
```diff
@@ -282,6 +282,60 @@ and :phpclass:`DOMNode` objects:
 
         $html = $crawler->html();
 
+Expression evaluation
+~~~~~~~~~~~~~~~~~~~~~
+
+The ``evaluate()`` method evaluates the given XPath expression.
+The return value depends if the expression operates on simple values
+(like html attributes), or returns a subset of the current document.
```
Comment (javiereguiluz) [inline components/dom_crawler.rst]: `html attributes` -> `HTML attributes`

**Review by javiereguiluz — APPROVED** [review]
👍

File: components/dom_crawler.rst (hunk 1 above)
Comment (jakzal) [inline components/dom_crawler.rst]: Fixed.

**Review by mickaelandrieu — APPROVED** [review]
Clear and complete ! Thank you :)

File: components/dom_crawler.rst [hunk 2]
```diff
@@ -282,6 +282,60 @@ and :phpclass:`DOMNode` objects:
 
         $html = $crawler->html();
 
+Expression evaluation
```
Comment (xabbuh) [inline components/dom_crawler.rst]: Evaluation

File: components/dom_crawler.rst [hunk 3]
```diff
@@ -282,6 +282,60 @@ and :phpclass:`DOMNode` objects:
...
+~~~~~~~~~~~~~~~~~~~~~
+
+The ``evaluate()`` method evaluates the given XPath expression.
+The return value depends if the expression operates on simple values
```
Comment (xabbuh) [inline components/dom_crawler.rst]: depends on?

File: components/dom_crawler.rst [hunk 4]
```diff
@@ -282,6 +282,60 @@ and :phpclass:`DOMNode` objects:
 
         $html = $crawler->html();
 
+Expression Evaluation
+~~~~~~~~~~~~~~~~~~~~~
+
+The ``evaluate()`` method evaluates the given XPath expression.
+The return value depends on if the expression operates on simple values
+(like HTML attributes), or returns a subset of the current document.
```
Comment (wouterj) [inline components/dom_crawler.rst]: "[...] if the expression operates on simple values (like HTML attributes) or a subset of the current document." ?

File: components/dom_crawler.rst [hunk 5]
```diff
@@ -282,6 +282,60 @@ and :phpclass:`DOMNode` objects:
...
+If the expression evaluates to a scalar value, an array of results will be
+returned. If the expression evaluates to a DOM document, a new ``Crawler``
+instance will be returned.
+
+This behavior is best illustrated with examples:
+
+.. code-block:: php
```
Comment (wouterj) [inline components/dom_crawler.rst]: please transform these 3 lines to `This behaviour is best illustrated with examples::`

**Comment by wouterj** [comment]
👍 
status: reviewed

File: components/dom_crawler.rst (hunk 4 above)
Comment (jakzal) [inline components/dom_crawler.rst]: Much better!

**Comment by wouterj** [comment]
Thanks @jakzal! I've done some rewrites while merging in 67706f6 to remove some duplication in this paragraph.

**Pull request merged**